src/events/cd-event-list.vue

```diff
@@ -38,18 +43,38 @@
             this.currentUser = response.body.user;
           });
       },
+      loadUsersProfile() {
+        return UserService.userProfileData(this.currentUser.id)
+          .then((response) => {
+            this.usersProfile = response.body;
+          });
+      },
+      loadUserDojoRole() {
+        return DojosService.getUsersDojos(this.currentUser.id, this.dojo.id)
+        .then((response) => {
+          this.usersDojos = response.body;
+        });
+      },
       loadEvents() {
         service.loadEvents(this.dojo.id).then((response) => {
           this.events = response.body;
         });
       },
+      async joinTheDojo() {
+        const userType = UsersUtil.isYouthOverThirteen(new Date(this.usersProfile.dob)) ? 'attendee-o13' : 'parent-guardian';
+        await DojosService.joinDojo(this.currentUser.id, this.dojo.id, [userType]);
+        /* eslint-disable no-alert */
+        alert('Congratulations, you have now joined the Dojo.');
+        /* eslint-enable no-alert */
+        this.loadUserDojoRole();
+      },
     },
     watch: {
       currentUser(newUser) {
         if (newUser) {
-          DojosService.getUsersDojos(newUser.id, this.dojo.id).then((response) => {
-            this.usersDojos = response.body;
-          });
+          const userDojoPromise = this.loadUserDojoRole();
+          const userProfilePromise = this.loadUsersProfile();
+          Promise.all([userDojoPromise, userProfilePromise]);
```

Comment:
What's the purpose of this `Promise.all`? It's not returned and there's no `then`. It's unnecessary.

Reply:
fair point